test/fulfillment_order_fulfillment_test.rb

```diff
@@ -0,0 +1,84 @@
+require 'test_helper'
+
+class FulfillmentOrderFulfillmentTest < Test::Unit::TestCase
+  def setup
+    super
+    fake "fulfillment_orders/519788021", method: :get,
+         body: load_fixture('fulfillment_order')
+  end
+
+  context "FulfillmentOrderFulfillment" do
+    context "#create" do
+      should "be able to create a fulfillment order fulfillment" do
+        create_fulfillment_attributes = {
+          message: "The message for this FO fulfillment",
+          notify_customer: true,
+          tracking_info: {
+            number: "XSDFHYR23475",
+            url: "https://tracking.example.com/XSDFHYR23475",
+            company: "TFTC - the fulfillment/tracking company"
+          },
+          line_items_by_fulfillment_order: [
+            {
+              fulfillment_order_id: 3,
+              fulfillment_order_line_items: [{ id: 2, quantity: 1 }]
+            }
+          ]
+        }
+        stub_request(:post, "https://this-is-my-test-shop.myshopify.com/admin/api/2019-01/fulfillments.json").with(
+          body: {
+            fulfillment: create_fulfillment_attributes
+          }
+        ).to_return(status: 200, body: ActiveSupport::JSON.encode(create_fulfillment_attributes.merge(id: 346743624)),
+                    headers: { content_type: "text/x-json" })
+
+        fulfillment = ShopifyAPI::FulfillmentOrderFulfillment.new(create_fulfillment_attributes)
+
+        saved = fulfillment.save
+
+        assert_equal true, saved
+        assert_equal 346743624, fulfillment.id
+      end
+    end
+
+    context "#update_tracking" do
+      should "be able to update tracking info for a fulfillment" do
+        tracking_info = {
+          number: 'JSDHFHAG',
+          url: 'https://example.com/fulfillment_tracking/JSDHFHAG',
+          company: 'ACME co',
+        }
+        fake_f = ActiveSupport::JSON.decode(load_fixture('fulfillment').gsub(/1Z2345/, 'JSDHFHAG'))['fulfillment']
```

Comment:
We should probably assert that the fixture's tracking number, url, and company are not the ones that we are updating to initially.

Reply:
There's only one request: `update_tracking` so we're only testing the serialization/deserialization of bodies and matching of the mocked request. I can update the test to be more explicit with the relevant fields.

Reply:
My thinking was along the lines of if the initial tracking info of the fulfillment is the same as the ones we are trying to update to (aka no change), then the update method could essentially do nothing (not make any network request), or the deserialization of the response messes up somewhere, and this test would still pass.

So by doing a quick assertion that the initial values indeed differ from the values that we are trying to change, we know for sure that 1. the stub was hit and 2. the deserialization did not fail.

Reply:
I see what you mean. That's guaranteed by [initializing the requesting fulfillment with only the id](https://github.com/Shopify/shopify_api/pull/639/files#diff-dcdc368f3229b0031925b1881ac52bdfR58-R59)